tests/pytests/test_index_oom.py

```diff
@@ -513,3 +512,490 @@ def test_oom_100_percent(env):
   bgIndexingStatusStr = "background indexing status"
   OOMfailureStr = "OOM failure"
   env.assertEqual(error_dict[bgIndexingStatusStr], OOMfailureStr)
+
```

Comment:
Great tests! 
It seems like there is some repeating code in the tests. Can we move some of it to helper functions?

Reply:
I tried to "compress" as much as possible without reducing readability